test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -14,6 +20,60 @@ async function tests() {
       AsyncIteratorPrototype);
   }
 
+  {
+    // v1 stream
+
+    const stream = new Stream();
+    stream.destroy = common.mustCall();
+    process.nextTick(() => {
+      stream.emit('data', 'asd0');
+      stream.emit('data', 'asd1');
+      stream.emit('end');
+    });
+
+    const iter = Readable.prototype[Symbol.asyncIterator].call(stream);
+    assert.strictEqual((await iter.next()).value, 'asd0');
+    for await (const d of iter) {
+      assert.strictEqual(d, 'asd1');
+    }
+  }
+
+  {
+    // v1 stream error
+
+    const stream = new Stream();
+    stream.close = common.mustCall();
+    process.nextTick(() => {
+      stream.emit('data', 0);
+      stream.emit('data', 1);
+      stream.emit('error', new Error('asd'));
+    });
+
+    const iter = Readable.prototype[Symbol.asyncIterator].call(stream);
```

Comment:
I don't think this is an expectable public API for this feature. Should we add it to `Stream` itself?

Reply:
This was mostly for easier use in https://github.com/nodejs/node/pull/31223.

I think it's a great idea to add it to `Stream` though.

Reply:
Fixed